scripts/fetch-user-profile-properties/README.md

```diff
@@ -116,8 +116,76 @@ Function GetUserProfileProperties($username) {
 StartProcessing
 
 ```
+
 [!INCLUDE [More about PnP PowerShell](../../docfx/includes/MORE-PNPPS.md)]
-***
+
+# [CLI for Microsoft 365 with PowerShell](#tab/cli-m365-ps)
+
+```powershell
+# Usage example:
+#   .\Fetch-User-Profile-Properties.ps1 -WebUrl "https://contoso.sharepoint.com/sites/Intranet"
+
+[CmdletBinding()]
+param (
+  [Parameter(Mandatory = $true, HelpMessage = "Web url from which to retrieve the users, e.g. https://contoso.sharepoint.com/sites/Intranet")]
+  [string]$WebUrl
+)
+begin {
+  $m365Status = m365 status
+  if ($m365Status -eq "Logged Out") {
+    m365 login
+  }
+  Write-Host "Connection Successful!" -f Green 
+}
+process {
+  #variables
+  $basePath = "D:\" #base path where you want to save CSV file("D:\Scripts\Output\...")
+
+  $userDetailsArray = @()
+  $dateTime = "{0:MM_dd_yy}_{0:HH_mm_ss}" -f (Get-Date)
+  $csvPath = $basePath + "userdetails-" + $dateTime + ".csv"
+
+  Write-Host "`nRetrieving users from web $WebUrl" -ForegroundColor Cyan
+  Write-Progress -Activity "Obtaining user details" -Status "Obtaining users" -PercentComplete 0
+
+  $users = m365 spo user list --webUrl $WebUrl | ConvertFrom-Json
+  $usersWithMail = $users | Where-Object {$_.Email -ne ""}
+  $usersCount = $usersWithMail.Count
```

Comment:
here we have one small mistake which really is a 'special case' kind of thing 😉.
If we have only one user on the site this will not end up a collection but a single element. If I am not mistaken the `Where-Object` will not return a collection but a single object. Therefore the `Count` property of this object will be empty and as a result will get an empty .csv file but we should expecting a file with single user 🤔.
Do you think it would be worth checking this issue 🤔?

Reply:
Hi Adam, are you certain about this? I've just checked the code and I just see "return res.value", and no checks if it has more than one value. Also, as we are calling /_api/web/siteusers, I'm fairly certain that there will never be a single item being returned, as the system accounts are also in this API call. I've just tested this creating a new Communication site and this contained 4 users by default:
- My own user, used for creating the site
- NT Service\\spsearch
- SharePoint App
- SHAREPOINT\\system

For an O365 group, there will be even more groups, as the O365 groups (Owners, Members, Visitors) will always appear in this API call.

Do you think I should rewrite this? Or do you have an example on where we will only get one item from the API call? As it seems strange to me for this to occur.

Reply:
I am not talking about CLI command. This is of course perfect 😄😉.
I am talking about the behavior of powershell Where-Object when we do it on objects. lets consider this as explanation
```
PS C:\Workspace\CLI> $a = '{"Email":"test"}' | ConvertFrom-Json
PS C:\Workspace\CLI> $a

Email
-----
test


PS C:\Workspace\CLI> $b = '{"Email":""}' | ConvertFrom-Json
PS C:\Workspace\CLI> $data = @($a, $b) # I have an array of 2 objects and only one has an email
PS C:\Workspace\CLI> $test = $data | Where-Object {$_.Email -ne ""} # as a result I get a single object, not an array with one element
PS C:\Workspace\CLI> $test.count # this is null, my object does no have count property
PS C:\Workspace\CLI> $c = '{"Email":"test2"}' | ConvertFrom-Json
PS C:\Workspace\CLI> $data = @($a, $b, $c)
PS C:\Workspace\CLI> $test = $data | Where-Object {$_.Email -ne ""} # now I get an array of two objects
PS C:\Workspace\CLI> $test.count # the array has count
2
``` 

so as you may if we will have a single object with email in list the `$usersCount = $usersWithMail.Count` will be null not 1

Reply:
I see, that makes sense. I will change this later today. Thanks for the feedback 👍